samples/extensions/dynamic_multisample_rasterization/dynamic_multisample_rasterization.cpp

```diff
@@ -0,0 +1,931 @@
+/* Copyright (c) 2024, Mobica Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "dynamic_multisample_rasterization.h"
+
+#include "gltf_loader.h"
+#include "scene_graph/components/material.h"
+#include "scene_graph/components/mesh.h"
+#include "scene_graph/components/pbr_material.h"
+#include "scene_graph/components/sub_mesh.h"
+
+DynamicMultisampleRasterization::DynamicMultisampleRasterization()
+{
+	title = "DynamicState3 Multisample Rasterization";
+
+	set_api_version(VK_API_VERSION_1_2);
```

Comment:
How about Dynamic Rendering? In spec I read that extension was introduced in Vulkan 1.2.197.

Also if you check dynamic rendering sample it also have Vulkan version set on 1.2.
When I lower version to 1.1 bunch of new VL errors appear.

Reply:
What's the actual VL error you get? Dynamic rendering works fine with 1.1 if you meed the criteria described under https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_KHR_dynamic_rendering.html#_extension_and_version_dependencies. Maybe you were missing some of the extensions required when using 1.1?

Reply:
Vulkan 1.2 needs to be used indirectly because of dynamic rendering (DR) extension. DR didn't need to use 1.2 but we need to do image layout transitions manually. 
If you check in spec: https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#VkImageLayout
when I want to use `VK_IMAGE_LAYOUT_DEPTH_ATTACHMENT_OPTIMAL` as new image layout I need to utilize Vulkan 1.2.

So to do proper image layout transitions we need to use Vulkan 1.2.

Reply:
Okay, that makes sense :)